semgrep/semgrep/rule.py

```diff
@@ -261,9 +261,25 @@ def sarif_tags(self) -> Iterator[str]:
         Tags to display on SARIF-compliant UIs, such as GitHub security scans.
         """
         if "cwe" in self.metadata:
-            yield "cwe"
+            # Handling unexpected formatting of CWE meta-data
```

Comment:
Hi @hunt3rkillerz, thanks for the PR. Is it reasonable to simply return the string from the metadata if it exists? This avoids all the heavy string parsing. E.g.

```python
if "cwe" in self.metadata:
    yield self.metadata['cwe']
if "owasp" in self.metadata:
    yield f"OWASP-{self.metadata['owasp']}"
```

Note that I prepended OWASP because our convention hasn't included it, but has for CWE:

```
javascript/lang/security/audit/path-traversal/path-join-resolve-traversal.yaml
47:    owasp: 'A5: Broken Access Control'
48:    cwe: 'CWE-22: Improper Limitation of a Pathname to a Restricted Directory (Path Traversal)'

javascript/lang/security/audit/non-constant-sql-query.yaml
10:    owasp: 'A1: Injection'
11:    cwe: "CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')"

javascript/lang/security/audit/unknown-value-with-script-tag.yaml
14:    owasp: 'A7: Cross-site Scripting (XSS)'
15:    cwe: "CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
```

Reply:
Hi @mschwager, the reason I tried to cut down the tags is I noticed that there was some inconsistent formatting in the rules repo (some rules had the full description and some did not). 

Additionally, I'm not sure if there is a string length limit on tags when you upload a SARIF to Github's API (happy to test this if the preference is to avoid the string parsing).

Reply:
I think simply returning what's in the metadata key (e.g. `self.metadata['cwe']`) would be easiest. If there are inconsistencies, or we want to improve the metadata, let's do so in the rules themselves instead of accounting for all cases here. If you have a list of rules whose metadata could be improved I'd be happy to help out :+1:

Reply:
The changes above have been implemented.

Reply:
Regarding rules with Metadata issues. I can probably go through the rules repo and see about cleaning up some formatting.

Reply:
Awesome, thank you!